test/SymbolExpressionGtest.cpp

```diff
@@ -0,0 +1,196 @@
+ */
+TEST(SymbolExpressionIntTestSuite, WhenConstructorIsCalled_ThenNoExceptionsAreThrown)
+{
+    try
+    {
+        calculator::expression::SymbolExpression<int> result{
+                SymbolExpressionTestConstants::GLYPH_X};
+    }
+    catch(const std::exception& e)
+    {
+        ADD_FAILURE() << "SymbolExpression was not successfully constructed";
+    }
+}
+
+
+
+/**
+ * @brief tests the evaluation of a SymbolExpression<int>, which should not 
+ * yield a value, as it has not had a value bound to it
+ */
+TEST(SymbolExpressionIntTestSuite, WhenCalculateExpressionIsCalled_ThenNoneIsReturned)
+{
+    calculator::expression::SymbolExpression<int> objectToTest{
+            SymbolExpressionTestConstants::GLYPH_X};
+    auto result = objectToTest.calculateExpression();
+    ASSERT_EQ(result, boost::none);
+}
+
+
+
+/**
+ * @brief tests binding a value to a symbol that matches the symbol of a
+ * SymbolExpression<int>, which should return a new ExpressionInterface<int>
+ * with the correct bound value
+ */
+TEST(SymbolExpressionIntTestSuite, WhenBindValueToSymbolIsCalled_ThenAnExpressionIsReturnedThatYieldsTheCorrectValue)
+{
+    calculator::expression::SymbolExpression<int> objectToTest{
+            SymbolExpressionTestConstants::GLYPH_X};
+    auto result = objectToTest.bindValueToSymbol(
+            SymbolExpressionTestConstants::GLYPH_X,
+            SymbolExpressionTestConstants::INT_VALUE_TO_BIND);
+    ASSERT_NE(result, nullptr);
+    auto value = result->calculateExpression();
+    ASSERT_NE(value, boost::none);
+    ASSERT_EQ(*value, SymbolExpressionTestConstants::INT_VALUE_TO_BIND);
+}
+
+
+
+/**
+ * @brief tests binding a value to a symbol that does not match the symbol of a
+ * SymbolExpression<int>, which should return a nullptr, indicating that
+ * there were no changes to this expression
+ */
+TEST(SymbolExpressionIntTestSuite, WhenBindValueToSymbolIsCalledWithoutTheMatchingGlyph_ThenNullptrIsReturned)
+{
+    calculator::expression::SymbolExpression<int> objectToTest{
+            SymbolExpressionTestConstants::GLYPH_X};
+    auto result = objectToTest.bindValueToSymbol(
+            SymbolExpressionTestConstants::GLYPH_Y,
+            SymbolExpressionTestConstants::INT_VALUE_TO_BIND);
+    ASSERT_EQ(result, nullptr);
+}
+
+
+
+/**
+ * @brief tests converting a SymbolExpression<int> to a string, which should 
+ * yield the glyph used to construct it
+ */
+TEST(SymbolExpressionIntTestSuite, WhenToStringIsCalled_ThenCorrectStringRepresentationIsReturned)
+{
+    calculator::expression::SymbolExpression<int> objectToTest{
+            SymbolExpressionTestConstants::GLYPH_X};
+    auto result = objectToTest.toString();
+    ASSERT_EQ(result, std::string(1, SymbolExpressionTestConstants::GLYPH_X));
+}
+
+
+
+/**
+ * @brief tests the construction of a SymbolExpression<double> with a known
+ * glyph
+ */
+TEST(SymbolExpressionDoubleTestSuite, WhenConstructorIsCalled_ThenNoExceptionsAreThrown)
+{
```

Comment:
not sure about this try catch, maybe checkout Jeff's or my calculator prs for gtest formatting